src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultKeyValueAdapter.cs

```diff
@@ -102,5 +92,34 @@ public bool NeedsRefresh()
         {
             return _secretProvider.ShouldRefreshKeyVaultSecrets();
         }
+
+        private string ParseSecretReferenceUri(ConfigurationSetting setting)
+        {
+            try
+            {
+                var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
+
+                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
```

Comment:
What if the read occurs and it isn't a start object? Seems there should be an exception?

Reply:
Right, had it in feature management adapter and forgot to match that logic here